File: PriceMovement/ClientApp/src/app/domain/underlying.ts
export class UnderlyingRecord {

  constructor(data: any) {
    this.portfolio = data.portfolio;
    this.securityId = data.securityId;

    this.linked = data.linked;

    this.price = data.price;
    this.price1d = data.price1d;
    this.priceDate = this.setDate(data.priceDate);
    this.priceDate1d = this.setDate(data.priceDate1d);

    this.businessDaysSincePriceDate = data.businessDaysSincePriceDate;
    this.portfolioNavCustom = data.portfolioNavCustom;
  }

  public portfolio: string;
  public securityId: string;
  public description: string;
  public priceCurrency: string;
  public price: number;
  public priceDate: Date;
  public businessDaysSincePriceDate: number;
  public price1d: number;
  public priceDate1d: Date;
  public portfolioNavCustom: number;

  public linked: string;
  public displayFactor: number;
  public factor: number;
  public factor1d: number;

  public get staleException() {
    return this.businessDaysSincePriceDate > 5 ? 'Stale Price - Over 5 Days' : '';
  }

  public setDate(date: any): Date | any {
    const temp = new Date(date);
    if (temp.getFullYear() !== 1970) {
      return temp;
    }

    return '';
  }
}

File: PriceMovement.Domain/Underlying.cs
namespace PriceMovement.Domain
{
    using System;

    /// <summary>
    /// The underlying record.
    /// </summary>
    public class UnderlyingRecord
    {
        /// <summary>
        /// Gets or sets the Portfolio.
        /// </summary>
        public string Portfolio { get; set; }

        /// <summary>
        /// Gets or sets the PortfolioId.
        /// </summary>
        public decimal PortfolioId { get; set; }

        /// <summary>
        /// Gets or sets the PortfolioNavCustom.
        /// </summary>
        public decimal PortfolioNavCustom { get; set; }

        /// <summary>
        /// Gets or sets the SecurityId.
        /// </summary>
        public string SecurityId { get; set; }

        /// <summary>
        /// Gets or sets the LinkedSecurity.
        /// </summary>
        public string LinkedSecurity { get; set; }

        /// <summary>
        /// Gets or sets the Price.
        /// </summary>
        public decimal? Price { get; set; }

        /// <summary>
        /// Gets or sets the  PriceDate.
        /// </summary>
        public DateTime? PriceDate { get; set; }

        /// <summary>
        /// Gets or sets the PriceCurrency.
        /// </summary>
        public string PriceCurrency { get; set; }

        /// <summary>
        /// Gets or sets the BusinessDaysSincePriceDate.
        /// </summary>
        public decimal BusinessDaysSincePriceDate { get; set; }
    }
}
